pipelines.md

```diff
@@ -0,0 +1,323 @@
+
+##### Dynamic volume provisioning
+
+If you run your cluster on a public cloud, you can set up a dynamic persistent volume by using your cloud provider’s default storage class. For an example of how to use dynamic persistent volume provisioning, see [Dynamic volume provisioning](https://github.com/kabanero-io/kabanero-pipelines/blob/master/docs/VolumeProvisioning.md#dynamic-volume-provisioning).
+
+#### Creating secrets
+
+Git secrets must be created in the `kabanero` namespace and associated with the service account that runs the pipelines. To configure secrets using the pipelines dashboard, see [Create secrets](https://kabanero.io/docs/ref/general/configuration/tekton-webhooks.html#create-secrets).
+
+Alternatively, you can [configure secrets in the Kubernetes console or set them up by using the Kubernetes CLI](https://docs.okd.io/latest/dev_guide/secrets.html#creating-secrets).
+
+### Running pipelines by using the pipelines dashboard webhook extension
+
+You can use the [pipelines dashboard webhook extension](https://github.com/tektoncd/experimental/blob/master/webhooks-extension/docs/GettingStarted.md) to drive pipelines that automatically build and deploy an application whenever you update the code in your Git repo. Events such as commits or pull requests can be set up to automatically trigger pipeline runs.
+
+### Running pipelines by using a script
+
+If you are developing a new pipeline and want to test it in a tight loop, you might want to use a script or manually drive the pipeline.
+
+1. Log in to your cluster. For example,
+
+   ```shell
+   oc login <master node IP>:8443
+   ```
+
+1. Clone the pipelines repo
+
+   ```shell
+   git clone https://github.com/kabanero-io/kabanero-pipelines
+   ```
+
+1. Run the following script with the appropriate parameters
+
+   ```shell
+   cd ./pipelines/sample-helper-files/./manual-pipeline-run-script.sh -r [git_repo of the Appsody project] -i [docker registery path of the image to be created] -c [application stack name of which pipeline to be run]"
+   ```
+
+   - The following example is configured to use the dockerhub container registry:
+
+      ```shell
+       ./manual-pipeline-run-script.sh -r https://github.com/mygitid/appsody-test-project -i index.docker.io/mydockeid/my-java-microprofile-image -c java-microprofile"
+      ```
+
+   - The following example is configured to use the local OpenShift container registry:
+
+      ```shell
+       ./manual-pipeline-run-script.sh -r https://github.com/mygitid/appsody-test-project -i docker-registry.default.svc:5000/kabanero/my-java-microprofile-image -c java-microprofile"
+      ```
+
+### Running pipelines manually from the command line
```

Comment:
Please add a lead in sentence as header followed directly by list items do not format well in the markdown build. Something like ......

Follow these steps to run a pipeline directly from the command line:

Reply:
Okay